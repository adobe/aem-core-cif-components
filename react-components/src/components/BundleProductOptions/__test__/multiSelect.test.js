/*******************************************************************************
 *
 *    Copyright 2020 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import MultiSelect from '../multiSelect';
import { render } from '@testing-library/react';

describe('<MultiSelect>', () => {
    const requiredItem = {
        option_id: 1,
        required: true
    };

    const sortedOptions = [
        {
            id: 1,
            quantity: 1,
            position: 1,
            is_default: true,
            price: 0,
            price_type: 'FIXED',
            can_change_quantity: false,
            label: 'Carmina Necklace',
            product: { id: 1355, name: 'Carmina Necklace', sku: 'VA13-GO-NA', __typename: 'SimpleProduct' },
            __typename: 'BundleItemOption'
        },
        {
            id: 2,
            quantity: 1,
            position: 2,
            is_default: false,
            price: 0,
            price_type: 'FIXED',
            can_change_quantity: true,
            label: 'Augusta Necklace',
            product: { id: 1356, name: 'Augusta Necklace', sku: 'VA14-SI-NA', __typename: 'SimpleProduct' },
            __typename: 'BundleItemOption'
        }
    ];

    const customization = [
        {
            id: 1,
            quantity: 1
        }
    ];

    const handleSelectionChange = jest.fn();

    it('renders the component', () => {
        const { asFragment } = render(
            <MultiSelect
                item={requiredItem}
                sortedOptions={sortedOptions}
                customization={customization}
                handleSelectionChange={handleSelectionChange}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
