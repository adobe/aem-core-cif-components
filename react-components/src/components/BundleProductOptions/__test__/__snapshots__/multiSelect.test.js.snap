// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSelect> renders the component 1`] = `
<DocumentFragment>
  <select
    class="bundleProduct__option field__input"
    multiple=""
    name="1"
    size="5"
  >
    <option
      value="1"
    >
      Carmina Necklace
    </option>
    <option
      value="2"
    >
      Augusta Necklace
    </option>
  </select>
</DocumentFragment>
`;
